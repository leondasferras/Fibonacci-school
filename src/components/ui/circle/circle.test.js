import renderer from "react-test-renderer";
import { Circle } from "./circle";


describe("Тест рендера компонента Circle:", ()=> {
  it("Circle без буквы рендерится без ошибок", () => {
    const tree = renderer.create(<Circle/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Circle с буквами рендерится без ошибок", () => {
    const tree = renderer.create(<Circle letter={'ab'}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Circle с текстом в head рендерится без ошибок", () => {
    const tree = renderer.create(<Circle head={'ab'}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Circle с react-элементом в head рендерится без ошибок", () => {
    const tree = renderer.create(<Circle head={
      <Circle isSmall={true} letter={'ab'} />
    }/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Circle с текстом в tail рендерится без ошибок", () => {
    const tree = renderer.create(<Circle tail={"abc"}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Circle с react-элементом в tail рендерится без ошибок", () => {
    const tree = renderer.create(<Circle tail={
      <Circle isSmall={true} letter={'ab'} />
    }/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Circle с index рендерится без ошибок", () => {
    const tree = renderer.create(<Circle index={5}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Circle с isSmall=true рендерится без ошибок", () => {
    const tree = renderer.create(<Circle isSmall={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Circle с состоянием default рендерится без ошибок", () => {
    const tree = renderer.create(<Circle state="default"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Circle с состоянием changing рендерится без ошибок", () => {
    const tree = renderer.create(<Circle state="changing"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Circle с состоянием modified рендерится без ошибок", () => {
    const tree = renderer.create(<Circle state="modified"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

})