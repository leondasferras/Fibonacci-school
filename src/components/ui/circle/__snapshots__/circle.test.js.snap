// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест рендера компонента Circle: Circle без буквы рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle: Circle с index рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    5
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle: Circle с isSmall=true рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle: Circle с react-элементом в head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          ab
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle: Circle с react-элементом в tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          ab
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Тест рендера компонента Circle: Circle с буквами рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ab
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle: Circle с состоянием changing рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle: Circle с состоянием default рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle: Circle с состоянием modified рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle: Circle с текстом в head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    ab
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест рендера компонента Circle: Circle с текстом в tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    abc
  </div>
</div>
`;
